data_adapter_oemof/calculations.py

```diff
@@ -34,3 +37,48 @@ def get_name(*args, counter=None):
 @calculation
 def get_capacity_cost(overnight_cost, fixed_cost, lifetime, wacc):
     return annuity(overnight_cost, lifetime, wacc) + fixed_cost
+
+
+def decommission(adapter_dict: dict) -> dict:
+    """
+
+    Takes adapter dictionary from adapters.py with mapped values.
+
+    Supposed to be called when getting default parameters
+
+    Non investment objects must be decommissioned in multi period to take end of lifetime
+     for said objet into account
+
+    Returns
+    dictionary (
+    -------
+
+    """
+    capacity_column = "capacity"
+    max_column = "max"
+
+    if capacity_column not in adapter_dict.keys():
+        logging.info("Capacity missing for decommissioning")
+        return adapter_dict
+
+    if not isinstance(adapter_dict[capacity_column], list):
+        logging.info("No capacity fading out that can be decommissioned.")
+        return adapter_dict
+
+    if max_column in adapter_dict.keys():
+        if adapter_dict[capacity_column] == adapter_dict[max_column]:
+            adapter_dict[max_column] = adapter_dict[capacity_column] / np.max(
+                adapter_dict[capacity_column]
+            )
+        else:
+            logging.info("Decommissioning and max value can not be set in parallel")
+            adapter_dict[max_column] = list(
+                (adapter_dict[max_column] / np.max(adapter_dict[capacity_column]))
+            )
+    else:
+        adapter_dict[max_column] = adapter_dict[capacity_column] / np.max(
+            adapter_dict[capacity_column]
+        )
```

Comment:
Can you explain whats happening in decommission function, as I don't quite get it from code alone?
Best to put in docstring directly...

Reply:
Baisicly I am just dividing the yealy changing capacity by the maximum capacity and then setting the max capacity as value for all years while using the quotient as value for `max`

I also wanted to check whether max value already is used and wheter it just is the same as the capacity value (as in data provided for steel industy)